pennylane/transforms/compile.py

```diff
@@ -191,7 +191,7 @@ def qfunc(x, y, z):
             def stop_at(obj):
                 return (obj.name in all_ops) and (not getattr(obj, "only_visual", False))
 
-            expanded_tape = tape.expand(stop_at=stop_at)
+            expanded_tape = tape.expand(stop_at=stop_at, depth=expand_depth)
```

Comment:
Please could you take a look at the docstrings for `compile`? Not due to this PR, but I think it's a bit confusing:

> If no basis set is specified, no expansion will be done.

But expansion is still done to "non-template" gates

> expand_depth (int) – When basis_set is specified, the depth to use for tape expansion into the basis gates.

But in this PR, `expand_depth` also comes into play when `basis_set` is not specified.

Reply:
While adding `depth=expand_depth` is one option to address the issue, I'm not convinced it's exactly what we're looking for here. Isn't it that we want to "expand until we get gates that aren't templates" but we should update that to "expand until we get gates that aren't templates and also get rid of adjoints".